src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -18,15 +19,19 @@ public class TagHelperContext
         /// Instantiates a new <see cref="TagHelperContext"/>.
         /// </summary>
         /// <param name="allAttributes">Every attribute associated with the current HTML element.</param>
+        /// <param name="parentItems">Items that have been created by parent <see cref="ITagHelper"/>s.</param>
         /// <param name="uniqueId">The unique identifier for the source element this <see cref="TagHelperContext" /> 
         /// applies to.</param>
         /// <param name="getChildContentAsync">A delegate used to execute and retrieve the rendered child content 
         /// asynchronously.</param>
-        public TagHelperContext([NotNull] IDictionary<string, object> allAttributes,
-                                [NotNull] string uniqueId,
-                                [NotNull] Func<Task<string>> getChildContentAsync)
+        public TagHelperContext(
+            [NotNull] IDictionary<string, object> allAttributes,
+            [NotNull] IDictionary<string, object> parentItems,
```

Comment:
or if `TagHelperExecutionContext` _must_ own the `Items` dictionary, expose a property of type `TagHelperExecutionContext` in this class and let tag helper authors get to the `Items` through that property.